### PR TITLE
Add AWSInfrastructureAccessRoleGrant type and resource under cluster

### PR DESCRIPTION
**Pull request by zgalor**
@jhernand @tzvatot PTAL

### CODE REVIEW

File: model/clusters_mgmt/v1/aws_infrastructure_access_role_grant_resource.model [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+		// Total number of items of the collection that match the search criteria,
+		// regardless of the size of the page.
+		out Total Integer
+
+		// Retrieved list of AWS infrastructure access role grants.
+		out Items []AWSInfrastructureAccessRoleGrant
+	}
+
+	// Create a new AWS infrastructure access role grant and add it to the collection of
```
Comment (jhernand) [inline model/clusters_mgmt/v1/aws_infrastructure_access_role_grant_resource.model]: Looks like you are using spaces to indent this, or maybe a mix of spaces and tabs. The rest of the code in this project uses only tabs; try to do that as well.

File: model/clusters_mgmt/v1/aws_infrastructure_access_role_grant_type.model [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	// Link to AWS infrastructure access role. 
+	link Role AWSInfrastructureAccessRole
+
+	// The user AWS IAM ARN we want to grant the role.
+	UserARN String
+
+	// State of the grant.
+	State AWSInfrastructureAccessState
+
+	// Description of the state
```
Comment (jhernand) [inline model/clusters_mgmt/v1/aws_infrastructure_access_role_grant_type.model]: End this sentence with a dot. That dot will be used by the tool that generates the documentation to separate sentences. The first sentence will be used as the summary of that generated documentation.

File: model/clusters_mgmt/v1/aws_infrastructure_access_state_type.model [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// State of an AWS incfrastructure access role grant.
+enum AWSInfrastructureAccessState {
```
Comment (jhernand) [inline model/clusters_mgmt/v1/aws_infrastructure_access_state_type.model]: This should probably be `AWSInfrastructureAccessRoleGrantState`.

File: model/clusters_mgmt/v1/aws_infrastructure_access_state_type.model [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	// The access role is in the progress of being granted.
+	InProgress
```
Comment (jhernand) [inline model/clusters_mgmt/v1/aws_infrastructure_access_state_type.model]: In other places when something is in progress we use `pending`. For example, we use `pending` when the cluster is waiting for an account claim. Can we do the same here?

File: model/clusters_mgmt/v1/cluster_resource.model [hunk 5]
```diff
@@ -64,4 +64,10 @@ resource Cluster {
 	locator Addons {
 		target AddOnInstallations
 	}
+
+	// Refrence to the resource that manages the collection of aws infrastructure
```
Comment (jhernand) [inline model/clusters_mgmt/v1/cluster_resource.model]: s/aws/AWS/

File: model/clusters_mgmt/v1/aws_infrastructure_access_state_type.model (hunk 4 above)
Comment (zgalor) [reply model/clusters_mgmt/v1/aws_infrastructure_access_state_type.model]: Those are the terms aws account operator uses. If we want `pending` this means I need to add a mapper, or our API can differ that our model, and the model can match account operator.
Anyway, I'll change it.

**Comment by zgalor** [comment]
@jhernand @tzvatot rePTAL

**Pull request merged**